include/libtorrent/config.hpp

```diff
@@ -518,6 +518,23 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #endif // TORRENT_HAS_SSE
 
+#if (defined __arm__ || defined __aarch64__)
+#define TORRENT_HAS_ARM 1
+#else
+#define TORRENT_HAS_ARM 0
+#endif // TORRENT_HAS_ARM
+
+#ifndef __has_builtin
+#define __has_builtin(x) 0  // for non-clang compilers
+#endif
+
+#if (TORRENT_HAS_SSE && __GNUC__) \
+	|| (TORRENT_HAS_ARM && defined __GNUC__ && !defined __clang__) \
+	|| (defined __clang__ && __has_builtin(__builtin_clz))
+#define TORRENT_HAS_BUILTIN_CLZ 1
```

Comment:
instead of having a complex condition here, what would you think of defining TORRENT_HAS_BUILTIN_CLZ right after TORRENT_HAS_SSE 1 is defined. then this can just be another condition under which to define it as 1. then at the end, if it's not defined, define it to 0 (a lot of other macros are defined this way, at the end they are defaulted if they haven't been set by any rule). Obviously it requires the conditions to set it to 1 to be mutually exclusive to avoid duplicate definitions.